README.md

```diff
@@ -1,2 +1,63 @@
-# Telegraf-Output-Apache-Arrow-Flight
-External Telegraf plugin for writing time series to Apache Arrow Flight
+# Apache Arrow Flight External Output Plugin
+
+This plugin is a general purpose output plugin for the [Apache Arrow Flight protocol](https://arrow.apache.org/docs/format/Flight.html).
+
+## Arrow Schema
+
+The plugin currently supports inserting data points with the following schema:
+
+```go
+schema:
+  fields: 3
+    - tid: type=int32
+    - timestamp: type=timestamp[ms]
+    - value: type=float32
+```
+
+Support for arbitrary schemas is planned.
+
+## Setting up and running the plugin
+
+To build the binary and run the plugin:
+
+1. Install the latest version of [Go](https://go.dev/doc/install).
+2. Build the binary:
+    * Windows: `go build -o binary/flight.exe cmd/main.go`
+    * Linux/macOS: `go build -o binary/flight cmd/main.go`
+3. Download the latest version of [telegraf](https://portal.influxdata.com/downloads/) for your platform. (To see which platform is needed, run the following command: `go env GOOS GOARCH`)
+4. Extract the telegraf executable and configuration to the repository folder.
+5. Configure the telegraf.conf file:
+   * In `telegraf.conf`, remove the comment in front of the tag `[[outputs.execd]]` and the option `command`.
+     * Windows: Assign the following to `command` so the resulting line becomes: `command = ["/path/to/flight.exe", "-config", "/path/to/sample.conf"]` 
+     * Linux/macOS: Assign the following to `command` so the resulting line becomes: `command = ["/path/to/flight", "-config", "/path/to/sample.conf"]` 
+   * Configure any input plugin to consume metrics. (the metric must adhere to the schema presented in this README)
+6. Configure the [sample configuration](\plugins\output\flight\sample.conf) to connect to the desired server, and designate the desired table to store metrics in.
+7. Run the plugin using telegraf: `telegraf --config telegraf.conf --input-filter chosen_input_plugin --output-filter execd`
+
+To run the tests: 
+1. Install the latest version of [Go](https://go.dev/doc/install).
+2. Start the Flight Server.
```

Comment:
Maybe write "Apache Arrow Flight" server since we generally include Apache in the name of Apache projects. Also if this can be an arbitrary flight server, maybe write "Start **an** Apache Arrow Flight server" and include "on port ????" if it must run on a specific port.

Reply:
I have fixed the naming. I also changed it to "Start an Apache Arrow Flight server", but the port just needs to match the configuration so I omitted that part.